llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -1366,6 +1367,8 @@ bool MemCpyOptPass::processMemSetMemCpyDependence(MemCpyInst *MemCpy,
   Value *SizeDiff = Builder.CreateSub(DestSize, SrcSize);
   Value *MemsetLen = Builder.CreateSelect(
       Ule, ConstantInt::getNullValue(DestSize->getType()), SizeDiff);
+  if (auto *SI = dyn_cast<SelectInst>(MemsetLen))
```

Comment:
Should this have a TODO or FIXME? I.e. is there a hope that we would be able to put something other than unknown here?

Reply:
Chatted offline with @xur-llvm, actually we do record a histogram of frequency of various kinds of sizes, so some kind of analysis would be possible. Let's see how many of these selects make it in CodeGenPrepare (we have the DEBUG_TYPE to help with that). My suspicion is that we (1) don't do a good VP propagation for memops in the first place, and (2) some of these memops are synthetic (from intrinsics). Currently the profcheck work is about branches (br, switch, select), but we should get back to this.